Mobile Design

jduong09 opened the pull request:
## Feature
- Pull Request for mobile responsive design on project.

jduong09 committed:
Added grid css responsiveness.

jduong09 committed:
Small work on design.

jduong09 committed:
Created separate table for mobile design.

jduong09 committed:
Made changes to use one table and adjust dates shown through media queries.

jduong09 committed:
Changed header css on table.

jduong09 committed:
moved CSS from app.css to specific css files.

jduong09 committed:
Small responsive changes.

jduong09 committed:
Finalized css for mobile. Opening for review.

jduong09 committed:
Made some changes.

jduong09 committed:
Finalized some css changes on response/admin page.

jduong09 commented:
### Update
- Focused on making the project mobile friendly.
- Added icons and hid description/location on mobile, which can be seen through clicking.
- Displayed description/location under header in desktop.

ziggysauce commented on root/client/src/components/poll/adminPage.js:
```diff
@@ -0,0 +1,134 @@
+        </div>
+        <div id="header-poll-btns">
+          <button id='btn-mobile-open' onClick={handleOpenDescription} ref={btnMobileOpen}>
+            <svg xmlns="http://www.w3.org/2000/svg" height="16" width="14" viewBox="0 0 448 512"><path d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z" fill="#FFFFFF"/></svg>
```
have you considered using a font library instead of manually hard-coding all of these svgs?

jduong09 replied:
I have not considered using a font library, i find hard-coding SVG's not a major issue. Is there an issue with it, something to worry about in your opinion?

ziggysauce replied:
for a small scale app, no, nothing to worry about. but for larger apps where usability and readability comes to mind, then using a library is often preferred. this is _generally_ the case in the workplace too, so I only suggested it to give you more familiarity of it in the future.

jduong09 replied:
For this project, I'll keep with the hard-code. For my next project, i'll make sure to use a font library. Thanks Dan.

ziggysauce commented on root/client/src/components/table/table.js:
```diff
@@ -117,12 +101,58 @@ export const Table = ({ pollAvailabilities, setPollAvailabilities, duration, sta
+    }
+  }
+
+  const handleNextDay = (e) => {
```
i've seen this logic a few times now. might be worth just making it a util function and referencing it where necessary

jduong09 replied:
The handleNextDay should only exist in one file, the table.js file. Let me know if i'm incorrect.

ziggysauce reviewed (commented):
assuming the description might be a longer body of text, you should use the `textarea` element instead of the `input` element

<img width="1073" alt="Screen Shot 2024-01-23 at 4 08 55 PM" src="https://github.com/jduong09/doodle/assets/25065329/11066cf6-8032-427e-bad6-eaf5faf5309c">

---

Put some padding/margin between the icons and text

<img width="1068" alt="Screen Shot 2024-01-23 at 4 10 07 PM" src="https://github.com/jduong09/doodle/assets/25065329/c4eaaa38-f6c4-4d45-86e9-5b82a4333f31">

---

In this view, i get that the content is hidden inside the hamburger menu, but it probably shouldnt be? there's clearly enough room on the screen to show it

https://github.com/jduong09/doodle/assets/25065329/86ad3df3-57d2-42cd-ac89-3e01e4f70572

---

Not sure what's going on here, but when i try to submit a response, none of the options are showing up

<img width="1075" alt="Screen Shot 2024-01-23 at 4 12 56 PM" src="https://github.com/jduong09/doodle/assets/25065329/5e73b2e2-8145-4e7f-89f4-48a0c15739bf">

---

Not necessary, but having any "submit" or "update" buttons can be sticky to the bottom of the mobile view. this makes for a better experience not having the user have to scroll all the way to the bottom just to submit if there is a very long list of options

<img width="1077" alt="Screen Shot 2024-01-23 at 4 14 01 PM" src="https://github.com/jduong09/doodle/assets/25065329/47d31974-ce19-4f51-a3eb-804371ee2431">

---

The "submissions" (the names of the people) look like buttons, but definitely are not, which can make for a bit of a confusing UX. I would instead make the names look more descriptive, and less button-like. Perhaps as a badge it would look less like a button?

<img width="1075" alt="Screen Shot 2024-01-23 at 4 15 08 PM" src="https://github.com/jduong09/doodle/assets/25065329/155d61fe-a111-4f90-b40c-f5b2384d4e9e">

jduong09 committed:
Merged main changes into current PR.

jduong09 committed:
Removed railway.toml file. Made changes based on PR review.

jduong09 commented:
## Update
### 1/29
- Changed poll description from input tag to textarea tag.
- Changed list item for peer's responses to be more badge-like.
- Added margin/padding to header description and location.

jduong09 committed:
Added sticky divs to adminPage and responsePage. Check comment for other changes on 1/30/24.

jduong09 committed:
Resolved merge conflicts with bug/formValidation branch.

jduong09 committed:
Resolved merge conflicts.

jduong09 committed:
Finished merge conflict.

jduong09 committed:
Deleted duplicate css code.

ziggysauce reviewed (commented):
for some reason, when I try creating a poll i'm seeing this error message and it's not being created

<img width="1326" alt="Screen Shot 2024-02-02 at 10 02 00 PM" src="https://github.com/jduong09/GetTogether/assets/25065329/c3eaee78-c961-45eb-987a-d2c10f4fd6b2">

jduong09 committed:
Programatically showed description expand button.

jduong09 committed:
Imported fonts.

jduong09 commented:
## Update
- PR is finished. Merging into main.
